app/next-client-app/api/projects.ts

```diff
@@ -0,0 +1,58 @@
+"use server";
+
+import request from "@/lib/api/request";
+import { fetchAllPages } from "@/lib/api/utils";
+
+const fetchKeys = {
+  list: (filter?: string) => (filter ? `projects/?${filter}` : "projects/"),
+  projectsDataset: (dataset: string) => `projects/?dataset=${dataset}`,
+  project: (id: string) => `projects/${id}/`,
+};
+
+export async function getProjectsList(
+  filter?: string | undefined
+): Promise<PaginatedResponse<Project>> {
```

Comment:
All these projects endpoints return the same type of `Project`, is this actually true? 
Looking at the api code it doesn't look like it.

For example `getProject` returns `ProjectSerializer` but the list view has 3 different serializer possibilities.
It's important we match up the return types to the API code accurately, so when those models are implemented on pages we can be safe that they actually contain that information.

Reply:
so only `ProjectDatasetSerializer`  will return (therefore require) a different type with other serialiser/API calls (I mean, `getProjectsList` `getProjectsDataset` `getProject` will both hit `ProjectWithMembersSerializer` ), but it hasn't been implemented yet, I suppose.
If what I am thinking is right, so maybe I only need to update the type here?